File: need4spd/euler_36.py
def isPalindrome(n):
    n_str = str(n)
    reverse_str = n_str[::-1]
    
    return n_str == reverse_str

result=0

for n in range(0, 1000001):
    n_2 = str(bin(n))[2:]
    
    if not isPalindrome(n):
        continue
    
    if n_2[0:1] == 0:
        continue
    
    if not isPalindrome(n_2):
        continue
    
    result+=n

print(result)

File: need4spd/euler_36.rb
def isPalidrome(n)
  n_str = n.to_s
  reverse_str = n_str.reverse
  
  return n_str == reverse_str
end

result=0

(0..1000000).each do |n|
  n_2 = n.to_s(2)
  
  unless isPalidrome(n)
    next
  end
  
  if n_2[0,1] == 0
    next
  end
  
  unless isPalidrome(n_2)
    next
  end
  
  result+=n
end

puts "#{result}"
